src/main/java/org/dsa/iot/alarm/OutOfRangeAlgorithm.java

```
@@ -46,27 +47,65 @@ protected String getAlarmMessage(AlarmWatch watch) {
         return String.format(pattern, String.valueOf(watch.getCurrentValue()));
     }
 
+    /**
+     * Try to extract a numeric from the value, even if it's another type.
+     *
+     * @return Double.NaN if a double can't be extracted.
+     */
+    private double getNumeric(Value value) {
+        Number num = value.getNumber();
+        if (num != null) {
+            return num.doubleValue();
+        }
+        Boolean bool = value.getBool();
+        if (bool != null) {
+            return bool ? 1 : 0;
+        }
+        String str = value.getString();
+        if (str != null) {
+            try {
+                return Double.parseDouble(str);
+            } catch (Exception possible) {
```

I'd add logging here instead of silencing the exception